### PR TITLE
Creating default Environment config files only when explicitly direct…

### PR DESCRIPTION
**Pull request by luketpeterson**
Creating default Environment config files only when explicitly directed.

This addresses https://github.com/trueagi-io/hyperon-experimental/issues/475

### CODE REVIEW

**Comment by vsbogd** [comment]
Luke, do you think there is a case when environment should create directory? In #475 I suggested that `modules` directory should be created by a tool which installs libraries,  `metta.init` script could be created manually (depends on its purpose). And in this case `Environment` could only provide the standard paths for this scripts and paths can be created by the code which uses `Environment` instances.

**Comment by luketpeterson** [comment]
> Luke, do you think there is a case when environment should create directory?

As we discussed on the call, my reasoning for keeping the default config creation inside the library is:

* The Environment needs to be read by the library.  So the logic to read it must be within the library
* The format of the Environment should be centralized in one place, so logic to update and modify the Environment belongs near logic to read it.

Also, it isn't big, so size isn't a factor in bloating the core library.

File: lib/src/metta/runner/environment.rs [hunk 1]
```diff
@@ -124,6 +126,17 @@ impl EnvBuilder {
...
+    pub fn create_config_dir(mut self) -> Self {
+        self.create_cfg_dir = true;
+        if self.no_cfg_dir {
+            panic!("Fatal Error: create_config_dir is incompatible with set_no_config_dir");
+        }
+        self
+    }
+
     /// Configures the Environment not to load nor create any config files
     pub fn set_no_config_dir(mut self) -> Self {
```
Comment (vsbogd) [inline lib/src/metta/runner/environment.rs:141]: I would also check that `self.no_cfg_dir` value doesn't conflict with `self.create_cfg_dir` value.

Comment (luketpeterson) [reply lib/src/metta/runner/environment.rs:141]: Added in [168231b](https://github.com/trueagi-io/hyperon-experimental/pull/478/commits/168231b61bdec8f1c80f368eceb86bd6669ea794)

**Comment by luketpeterson** [comment]
What additional changes are required to make this ready for merge?

**Review by vsbogd — APPROVED** [review]

**Comment by vsbogd** [comment]
Thanks Luke for fixing this!

**Pull request merged**